### ISSUE TITLE
The 3D attention mask in the LongFormer is wrong

### ISSUE THREAD

**Issue by syiswell** [huggingface/transformers#23044]
### Feature request

I used a 3d attention mask in the LongFormer, but also failed. I find that the code
attention_mask = nn.functional.pad(
attention_mask, (0, padding_len), value=0
) # no attention on the padding tokens
in line 1626 in modeling_longformer may not support the 3D attention mask. Please correct me if I am wrong

### Motivation

I want to input a 3d attention mask in the LongFormer to conduct the visible field for different tokens.

### Your contribution

posting a code snippet example
